spring-cloud-gcp-logging/src/main/java/org/springframework/cloud/gcp/logging/TraceIdFromRequestExtractor.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.logging;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * A type which can extract a trace ID from an HTTP request.
+ *
+ * @author Chengyuan Zhao
+ */
+public interface TraceIdFromRequestExtractor {
```

Comment:
I'm really fine just with the `TraceIdExtractor` name...

Reply:
I was thinking the same actually.